dipy/tracking/utils.py

```diff
@@ -483,13 +483,19 @@ def random_seeds_from_mask(mask, seeds_count=1, seed_count_per_voxel=True,
     else:
         seeds_per_voxel = seeds_count
 
-    # Generate as many random triplets as the number of seeds needed
-    grid = np.random.random([seeds_per_voxel * num_voxels, 3])
-    # Repeat elements of 'where' so that it can be added to grid
-    where = np.repeat(where, seeds_per_voxel, axis=0)
-    seeds = where + grid - .5
+    seeds = []
+    for i in range(1, seeds_per_voxel + 1):
+        for s in where:
+            # Set the random seed with the current seed, the current value of
+            # seeds per voxel and the global random seed.
+            np.random.seed((s + 1) * i + random_seed)
+            # Generate random triplet
+            grid = np.random.random(3)
+            seed = s + grid - .5
+            seeds.append(seed)
     seeds = asarray(seeds)
 
+    np.random.seed(random_seed)
     if not seed_count_per_voxel:
         # Randomize the seeds and select the requested amount
         np.random.shuffle(seeds)
```

Comment:
I'll modify the function to have this feature.

Reply:
But if we want less than len(seeds) we have not the same seeds. Something that could be correct this problem is to shuffle the data first and then compute the seeds. 

@gabknight Do you agree ?

Reply:
I will push this feature. I tested it and it works correctly for npv and nt

Reply:
OK. I think you solved it but by shuffling the indices instead of the seeds. 

Can you add a test comparing the resulting seed positions, with the same `random_seed` but various npv and nt, e.g. with a mask of 100 voxel:
```
random_seeds_from_mask(2, True)[:150] 
== random_seeds_from_mask(3, True)[:150] 
== random_seeds_from_mask(150, False)[:150] 
== random_seeds_from_mask(500, False)[:150]
```